CODE_STYLE.md

```diff
@@ -60,9 +59,37 @@
   all shared and private variables.
 * All critical sections must have a unique name.
 
+## Precision
+* Precision of all real arguments are explicitly defined as `real(kind=r4_kind)`,
+  `real(kind=r8_kind)`, or as any other precision parameters defined in platform_mod.
+* The precision of real numerical values should be consistent with the precision
+  of the associated variable.  For example, if the variable `a` has been declared
+  as r8_kind, then `a=1.4_r8_kind` is acceptable. The following, a=1.4 and a=(1.4,kind=r8_kind),
+  are not acceptable since the numerical value of 1.4 will be represented in the default precision
+  set by the compiler.
+* The precision of integers do not need to be explicitly defined and can be determined at compile time.
+* If the precision of integers are explicitly defined, they are defined with the precision parameters,
+  _e.g._ i4_kind, i8_kind, found in platform_mod.
+
+## Macros
+* All letters in the macro names are capitalized
+* All macro names end with an underscore "_"
+* All precision related macro names start with the letters "FMS"
+* Macro names should be unique to each module.  For example,
+  `FMS_AU_KIND_` is used in axis_utils_mod.
+  `FMS_HI_KIND_` is used in horiz_interp_mod
+
+## .fh and .inc files
+* The .fh header files contain macro definitions.
+* If the .fh files contain mainly precision related macro definitions, the files
+  should be named with `_r4.fh` and `_r8.fh` extensions in the include subdirectory found
+  in the module directory.  These .fh files are `#include`-ed at the end of the .F90 module files.
+* For precision related .inc files, the .inc files contain the procedure definitions and are
+  `#include`-ed at the end of both *_r4.fh and *_f8.fh files.  These .inc files are located in the
+  same include subdirectory as the .fh files.  See below for details.
```

Comment:
@uramirez8707 not sure if this is clear enough, what do you think?

Reply:
I think this + the example below is good enough.